### DISCUSSION TITLE
Querying all Product with a certain ObjectBrick

### DISCUSSION THREAD

**Discussion by julescournut** [pimcore/pimcore#16040] [Q&A / Support] [answered]
Hello all, 
I'm trying to retrieve all products that are linked to a certain ObjectBricks. 
I found [here](https://pimcore.com/docs/platform/Pimcore/Objects/Object_Classes/Data_Types/Object_Bricks/#querying-for-objectbrick-data) how to query data within objectbricks but here I just want ALL products that have this objectbricks linked no matter what data they have. 
How can I do that ? 
Thank you for your help.
Jules.

**Comment by NiklasBr** [comment, marked as answer]
Solved it something like this, it appears the class will only have a row in the table object_brick_query_... if it has the brick at all.

```php
$listing = new YourProductClass\Listing();
$listing->onCreateQueryBuilder(
    static function (QueryBuilder $queryBuilder) {
        $tableInfo = $queryBuilder->getQueryPart('from');
        $queryBuilder->innerJoin(
            fromAlias:  $tableInfo[0]['table'],
            join:       'object_brick_query_YOURTABLENAME',
            condition:  "object_brick_query_YOURTABLENAME.`o_id` = {$tableInfo[0]['table']}.`ooo_id`",
        );
    }
);
```

**Reply by julescournut** [reply]
Hello Niklas, thank you for your answer, it works after some adjustements.
I didn't specified my pimcore version but I'm in 11 so it will look like this :

``` php
$listing = new Product\Listing();
$listing->onCreateQueryBuilder(
    static function (QueryBuilder $queryBuilder) use ($alias) {
          $tableInfo = $queryBuilder->getQueryPart('from');
          $queryBuilder->innerJoin(
              fromAlias:  $tableInfo[0]['table'],
              join:       "object_brick_query_YOURTABLENAME",
              alias:      $alias,
              condition:  "$alias.`id` = {$tableInfo[0]['table']}.`oo_id`",
          );
    }
);
```
Thanks a lot !
